crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -338,10 +339,21 @@ def prepare_nodes_for_os_upgrade
       end
       if ready_for_reboot
         @logger.debug("Rebooting node #{node.name} for operating system upgrade")
-        node.ssh_cmd("/sbin/reboot")
+        ssh_status = node.ssh_cmd("/sbin/reboot")
+        if ssh_status[0] != 200
+          @logger.error("Upgrade failed for machine #{node.name}. Could not ssh.")
+          upgrade_nodes_failed.push(node.name)
+        end
+      else
+        @logger.error("Upgrade failed for #{node.name}. Node not ready for reboot")
+        upgrade_nodes_failed.push(node.name)
       end
     end
-    true
+    if upgrade_nodes_failed.any?
+      return false, upgrade_nodes_failed
+    else
+      return true
```

Comment:
@dguitarbite Right, and that's what we just mentioned as problematic. Because it's going to be difficult for the caller to test if the call was successful or not. Please either return ```nil``` or just the empty ```upgrade_nodes_failed``` list in case of success.